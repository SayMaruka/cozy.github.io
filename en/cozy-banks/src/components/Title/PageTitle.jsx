import React from 'react'
import PropTypes from 'prop-types'
import Title from 'components/Title/Title'
import { BarCenter } from 'components/Bar'
import useBreakpoints from 'cozy-ui/transpiled/react/hooks/useBreakpoints'
import styles from './PageTitle.styl'

export const BarTitle = ({ children }) => {
  return (
    <BarCenter>
      <Title className={styles.PageTitle}>{children}</Title>
    </BarCenter>
  )
}

const PageTitle = ({ children }) => {
  const { isMobile } = useBreakpoints()
  return isMobile ? <BarTitle>{children}</BarTitle> : <Title>{children}</Title>
}

PageTitle.propTypes = {
  children: PropTypes.node.isRequired
}

export default PageTitle
