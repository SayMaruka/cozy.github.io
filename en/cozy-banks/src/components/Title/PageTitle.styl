@require 'tools/mixins'
@require 'settings/breakpoints'

+small-screen()
    .PageTitle
        display     flex
        align-items center
        overflow hidden
