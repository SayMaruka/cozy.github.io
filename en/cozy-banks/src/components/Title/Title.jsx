import React from 'react'
import PropTypes from 'prop-types'
import cx from 'classnames'
import { MainTitle } from 'cozy-ui/transpiled/react/Text'
import styles from 'components/Title/Title.styl'
import useTheme from 'components/useTheme'

const Title = props => {
  const theme = useTheme()
  const { children, className } = props

  return (
    <MainTitle
      tag="h1"
      ellipsis={true}
      className={cx(styles.Title, styles[`TitleColor_${theme}`], className)}
    >
      {children}
    </MainTitle>
  )
}

Title.propTypes = {
  children: PropTypes.node.isRequired
}

export default Title
