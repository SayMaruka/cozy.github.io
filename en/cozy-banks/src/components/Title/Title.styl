@require 'settings/breakpoints'

.Title
    line-height 2.5rem
    margin 0 auto
    margin-bottom .5rem
    display flex

.TitleColor_default
    color var(--black) !important

.TitleColor_primary
    color var(--primaryContrastTextColor) !important

+small-screen()
    .Title
        font-size 1.125rem
        line-height 1.5rem
        margin-bottom 0
