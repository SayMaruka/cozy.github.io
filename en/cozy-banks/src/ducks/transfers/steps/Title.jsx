import React from 'react'
import { Title as UITitle, withBreakpoints } from 'cozy-ui/transpiled/react'

const _Title = ({ children, breakpoints: { isMobile } }) => {
  return (
    <UITitle className={'u-mb-1 ' + (isMobile ? 'u-ta-center' : '')}>
      {children}
    </UITitle>
  )
}

const Title = React.memo(withBreakpoints()(_Title))

export default Title
