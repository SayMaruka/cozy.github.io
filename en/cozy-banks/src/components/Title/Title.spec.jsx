import React from 'react'
import { shallow } from 'enzyme'
import Title from './Title'

describe('Title', () => {
  it(`should display children`, () => {
    expect(shallow(<Title>content</Title>).getElement()).toMatchSnapshot()
  })

  it(`should extend className`, () => {
    expect(
      shallow(<Title className="noPaddingBottom">content</Title>).getElement()
    ).toMatchSnapshot()
  })

  it(`should set color default`, () => {
    expect(
      shallow(<Title color="default">content</Title>).getElement()
    ).toMatchSnapshot()
  })

  it(`should set color primary`, () => {
    expect(
      shallow(<Title color="primary">content</Title>).getElement()
    ).toMatchSnapshot()
  })
})
