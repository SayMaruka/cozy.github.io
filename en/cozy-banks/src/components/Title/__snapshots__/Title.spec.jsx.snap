// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title should display children 1`] = `
<MainTitle
  className="Title TitleColor_default"
  ellipsis={true}
  tag="h1"
>
  content
</MainTitle>
`;

exports[`Title should extend className 1`] = `
<MainTitle
  className="Title TitleColor_default noPaddingBottom"
  ellipsis={true}
  tag="h1"
>
  content
</MainTitle>
`;

exports[`Title should set color default 1`] = `
<MainTitle
  className="Title TitleColor_default"
  ellipsis={true}
  tag="h1"
>
  content
</MainTitle>
`;

exports[`Title should set color primary 1`] = `
<MainTitle
  className="Title TitleColor_primary"
  ellipsis={true}
  tag="h1"
>
  content
</MainTitle>
`;
